crates/polars-core/src/chunked_array/logical/categorical/builder.rs

```diff
@@ -514,6 +538,42 @@ impl CategoricalChunked {
 
         CategoricalChunked::from_cats_and_rev_map_unchecked(cats, Arc::new(rev_map))
     }
+
+    /// Create a [`CategoricalChunked`] from a fixed list of categories and a List of strings.
+    /// This will error if a string is not in the fixed list of categories
+    pub fn from_utf_to_enum(
+        values: &Utf8Chunked,
+        categories: &Utf8Array<i64>,
+    ) -> PolarsResult<CategoricalChunked> {
+        polars_ensure!(categories.null_count()  == 0, ComputeError: "categories can not contain null values");
+
+        // Build a mapping string -> idx
+        let mut map = PlHashMap::with_capacity(categories.len());
+        for (idx, cat) in categories.values_iter().enumerate_idx() {
+            #[allow(clippy::unnecessary_cast)]
+            map.insert(cat, idx as u32);
+        }
+        // Find idx of every value in the map
+        let ca_idx: UInt32Chunked = values
+            .into_iter()
+            .map(|opt_s: Option<&str>| {
+                opt_s
+                    .map(|s| {
+                        map.get(s).copied().ok_or_else(
+                            || polars_err!(OutOfBounds: "value {} is not present in Enum: {:?}",s,categories),
+                        )
+                    })
+                    .transpose()
+            })
+            .collect::<Result<UInt32Chunked, PolarsError>>()?;
+        let rev_map = RevMapping::build_enum(categories.clone());
```

Comment:
nit: if we always clone the `categories`, I think we should accept an owned version in the funcion signature.